viewer/src/tools/HtmlOverlayTool.test.ts

```diff
@@ -0,0 +1,154 @@
+    htmlElement.style.cssText = 'position: absolute;';
+    const position = new THREE.Vector3();
+
+    // Act & Assert
+    expect(() => helper.add(htmlElement, position)).not.toThrow();
+  });
+
+  test('forceUpdate() places overlays correctly', () => {
+    // Arrange
+    const helper = new HtmlOverlayTool(viewer);
+    const htmlElement = document.createElement('div');
+    htmlElement.style.position = 'absolute';
+    expect(htmlElement.style.top).toBeEmpty();
+    expect(htmlElement.style.left).toBeEmpty();
+    const position = new THREE.Vector3(0, 0, 0.5);
+
+    // Act
+    helper.add(htmlElement, position);
+
+    // Assert
+    expect(htmlElement.style.visibility).toBe('visible');
+    expect(htmlElement.style.top).toBe(`${renderer.domElement.height / 2}px`);
+    expect(htmlElement.style.left).toBe(`${renderer.domElement.width / 2}px`);
+  });
+
+  test('Hides overlay if behind camera or behind far plane', () => {
+    // Arrange
+    const helper = new HtmlOverlayTool(viewer);
+    const behindCameraElement = document.createElement('div');
+    behindCameraElement.style.position = 'absolute';
+    const behindFarPlaneElement = document.createElement('div');
+    behindFarPlaneElement.style.position = 'absolute';
+
+    // Act
+    helper.add(behindCameraElement, new THREE.Vector3(0, 0, -1));
+    helper.add(behindFarPlaneElement, new THREE.Vector3(0, 0, 10));
+
+    // Assert
+    expect(behindCameraElement.style.visibility).toBe('hidden');
+    expect(behindFarPlaneElement.style.visibility).toBe('hidden');
+  });
+
+  test('Triggers position update callback', () => {
+    // Arrange
+    const helper = new HtmlOverlayTool(viewer);
+    const behindCameraElement = document.createElement('div');
+    behindCameraElement.style.position = 'absolute';
+    const behindFarPlaneElement = document.createElement('div');
+    behindFarPlaneElement.style.position = 'absolute';
+    const withinNearAndFarPlaneElement = document.createElement('div');
+    withinNearAndFarPlaneElement.style.position = 'absolute';
+    const options: HtmlOverlayOptions = { positionUpdatedCallback: jest.fn() };
+
+    // Act
+    helper.add(behindCameraElement, new THREE.Vector3(0, 0, -1), options);
+    helper.add(behindFarPlaneElement, new THREE.Vector3(0, 0, 10), options);
+    helper.add(withinNearAndFarPlaneElement, new THREE.Vector3(0, 0, 0.5), options);
+
+    // Assert
+    expect(options.positionUpdatedCallback).toHaveBeenCalledWith(
+      behindCameraElement,
+      expect.any(THREE.Vector2),
+      expect.any(THREE.Vector3),
+      expect.anything()
+    );
+    expect(options.positionUpdatedCallback).toHaveBeenCalledWith(
+      behindFarPlaneElement,
+      expect.any(THREE.Vector2),
+      expect.any(THREE.Vector3),
+      expect.anything()
+    );
+    expect(options.positionUpdatedCallback).toHaveBeenCalledWith(
+      withinNearAndFarPlaneElement,
+      expect.any(THREE.Vector2),
+      expect.any(THREE.Vector3),
+      expect.anything()
```

Comment:
😮  but you don't assert anything (pun not intended) here

https://jestjs.io/docs/en/expect#expectanything

Or am I just misunderstanding what it does here?

Reply:
it asserts that the function is called with `withinNearAndFarPlaneElement` as the first argument - i.e. it expects that the callback is triggered for all registered elements. I agree it was a bit cumbersome way to verify it though

Reply:
oh God, I totally missed `toHaveBeenCalledWith(`, sorry

Reply:
oh God, I just totally missed `toHaveBeenCalledWith(`, sorry